src/client/batch.rs

```diff
@@ -1,191 +1,106 @@
-use serde::Deserialize;
-
-use std::fs::File;
-
-use super::*;
-
+use csv;
 use rust_decimal::Decimal;
-use rust_decimal_macros::dec;
+use std::path::Path;
+use std::str::FromStr;
 
-#[derive(Debug, Deserialize)]
-pub struct PaymentInput {
-    pub username: String,
-    pub usd: Decimal,
-    pub memo: Option<String>,
-}
+use crate::client::GaloyClient;
+use crate::client::Wallet;
 
-impl From<PaymentInput> for Payment {
-    fn from(input: PaymentInput) -> Payment {
-        Payment {
-            username: input.username,
-            usd: input.usd,
-            sats: None,
-            wallet_id: None,
-            memo: input.memo,
-        }
+// Utility function to check if file exists
+pub fn check_file_exists(file: &str) -> anyhow::Result<()> {
+    let file_path = Path::new(&file);
+    if !file_path.exists() {
+        return Err(anyhow::anyhow!("File not found: {}", file));
     }
+    Ok(())
 }
 
-#[derive(Debug)]
-struct Payment {
-    username: String,
-    usd: Decimal,
-    sats: Option<Decimal>,
-    wallet_id: Option<String>,
-    memo: Option<String>,
-}
-
-pub struct Batch {
-    payments: Vec<Payment>,
-    client: GaloyClient,
-    /// price in btc/usd
-    price: Decimal,
-}
-
-impl Batch {
-    pub fn new(client: GaloyClient, price: Decimal) -> Self {
-        let payments: Vec<Payment> = vec![];
-        Self {
-            payments,
-            client,
-            price,
-        }
+// Utility function to read and validate the CSV file
+pub fn validate_csv(
+    galoy_cli: &GaloyClient,
+    file: &str,
+) -> anyhow::Result<(Vec<csv::StringRecord>, Wallet)> {
+    let mut reader = csv::ReaderBuilder::new().delimiter(b',').from_path(file)?;
+    let headers = reader.headers()?.clone();
+
+    if &headers[0] != "username"
+        || (&headers[1] != "cents" && &headers[1] != "sats")
+        || (headers.len() == 3 && &headers[2] != "memo")
+    {
+        return Err(anyhow::anyhow!(
+            "CSV format not correct, requires: username, (cents or sats), memo(optional)"
+        ));
     }
 
-    pub fn add(&mut self, input: PaymentInput) {
-        self.payments.push(input.into());
-    }
-
-    pub fn add_csv(&mut self, filename: String) -> anyhow::Result<()> {
-        let file = File::open(filename)?;
-        let mut rdr = csv::Reader::from_reader(file);
-        for result in rdr.deserialize() {
-            let record: PaymentInput = result?;
-            self.add(record);
-        }
-
-        Ok(())
-    }
-
-    pub fn len(&self) -> usize {
-        self.payments.len()
-    }
-
-    pub fn is_empty(&self) -> bool {
-        self.payments.is_empty()
-    }
+    let wallet_type = if headers.get(1) == Some("cents") {
```

Comment:
are we assuming that it we sent sats, it's from the btc wallet, and if we send cents, it's from the usd wallet?

I don't think we should make this assumption. we can send "100 dollar worth of bitcoin" from the bitcoin wallet.

Reply:
I think typically people right not send "cents" or "cents equivalent", so we could start with this one.

Reply:
Batch payments is built upon intraledger payments, which does not support sending cents from a Bitcoin wallet or sats from a USD wallet. Due to this, our current implementation assumes that if the amount is in cents, it is sent from the USD wallet and vice versa. Once we incorporate the capability for conversions in intraledger payments, we will extend the batch payments feature to allow this.